### PR TITLE
cmake: Allow requiring minimum versions using sdk-os-version triples

### PR DESCRIPTION
**Pull request by erg**

### CODE REVIEW

**Comment by erg** [comment]
@swift-ci Please smoke test

**Comment by erg** [comment]
@swift-ci Please smoke test OS X platform

**Comment by erg** [comment]
@swift-ci Please smoke test OS X platform

File: cmake/modules/AddSwift.cmake [hunk 1]
```diff
@@ -301,7 +317,10 @@ function(_add_variant_link_flags)
     ENABLE_ASSERTIONS "${LFLAGS_ENABLE_ASSERTIONS}"
     ENABLE_LTO "${LFLAGS_ENABLE_LTO}"
     ANALYZE_CODE_COVERAGE "${LFLAGS_ANALYZE_CODE_COVERAGE}"
-    DEPLOYMENT_VERSION_IOS "${LFLAGS_DEPLOYMENT_VERSION_IOS}"
+    DEPLOYMENT_VERSION_OSX "${CFLAGS_DEPLOYMENT_VERSION_OSX}"
+    DEPLOYMENT_VERSION_IOS "${CFLAGS_DEPLOYMENT_VERSION_IOS}"
+    DEPLOYMENT_VERSION_TVOS "${CFLAGS_DEPLOYMENT_VERSION_TVOS}"
+    DEPLOYMENT_VERSION_WATCHOS "${LFLAGS_DEPLOYMENT_VERSION_WATCHOS}"
```
Comment (gparker42) [inline cmake/modules/AddSwift.cmake]: Is "LFLAGS" a typo?

**Comment by gparker42** [comment]
What is the failure mode if one of the overlays builds for some OS without specifying a deployment target? Does it silently build with no deployment target or some default deployment target?

File: cmake/modules/AddSwift.cmake (hunk 1 above)
Comment (erg) [inline cmake/modules/AddSwift.cmake]: Yes, thank you for catching that.

**Comment by erg** [comment]
The overlays build for default targets that get overwritten by ``--darwin-sdk-deployment-targets=`` in this patch.

**Comment by erg** [comment]
@swift-ci Please smoke test

**Pull request merged**